src/game/client/videoplayer/videoplayer.h

```diff
@@ -0,0 +1,72 @@
+/**
+ * @file
+ *
+ * @author OmniBlade
+ *
+ * @brief Virtual interface for video players.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+
+#include "always.h"
+#include "asciistring.h"
+#include "ini.h"
+#include "subsysteminterface.h"
+#include <vector>
+
+class VideoStream;
+
+struct Video
+{
+    AsciiString file_name;
+    AsciiString internal_name;
+    AsciiString world_builder_comment;
+};
+
+class VideoPlayer : public SubsystemInterface
+{
+public:
+    VideoPlayer() : m_videosAvailableToPlay(), m_firstStream(nullptr) {}
+    virtual ~VideoPlayer();
+
+    virtual void Init() override;
+    virtual void Reset() override;
+    virtual void Update() override;
+
+    virtual void Deinit() {}
+    virtual void Lose_Focus() {}
+    virtual void Regain_Focus() {}
+    virtual int Open(AsciiString title) { return 0; }
+    virtual int Load(AsciiString title) { return 0; }
+    virtual VideoStream *First_Stream() { return m_firstStream; }
```

Comment:
You could make some methods here `const`

Reply:
I'll double check the original prototypes, its possible some are intended to be const, First_Stream should almost certainly be const from the look of it. The others will have more complex final implementations in a child class though, the original has an implementation built around the bink library which we'll almost certainly want to replace with the ffmpeg libs.